Objective-C/Tests/DatabaseTest.m

```diff
@@ -582,6 +582,57 @@ - (void) testSaveDocWithDeletedConflictUsingConcurrencyControl: (int)concurrency
     [self cleanDB];
 }
 
+- (void) testCancelConflictHandler {
+    NSString* docID = @"doc1";
+    CBLMutableDocument* doc = [[CBLMutableDocument alloc] initWithID: docID];
+    [doc setString: @"Tiger" forKey: @"firstName"];
+    [self saveDocument: doc];
+    AssertEqual([self.db documentWithID: docID].generation, 1u);
+    
+    CBLMutableDocument* doc1a = [[self.db documentWithID: docID] toMutable];
+    CBLMutableDocument* doc1b = [[self.db documentWithID: docID] toMutable];
+    
+    [doc1a setString: @"Scotty" forKey: @"nickName"];
+    [self saveDocument: doc1a];
+    AssertEqual([self.db documentWithID: docID].generation, 2u);
+    
+    NSError* error;
+    [doc1b setString: @"Scott" forKey: @"nickName"];
+    [self.db saveDocument: doc1b
+          conflictHandler:^BOOL(CBLMutableDocument * cur, CBLDocument * old) {
```

Comment:
The names `cur` and `old` confused me -- `old` isn't old, it's the newer revision in the database. So I had them backwards. Maybe `savingDoc` and `dbDoc` or something like that?

Reply:
As per the design spec this variables are `document` and `oldDocument`. We can discuss about this when Pasin is back.

Reply:
corrected to use the design-spec, with `document` and `oldDocument`